riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/testing/ExpectingHttpRequestFactory.java

```diff
@@ -0,0 +1,46 @@
+package org.zalando.riptide.spring.testing;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.AsyncClientHttpRequest;
+import org.springframework.http.client.AsyncClientHttpRequestFactory;
+import org.springframework.http.client.ClientHttpRequest;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.mock.http.client.MockAsyncClientHttpRequest;
+import org.springframework.test.web.client.RequestExpectationManager;
+
+import java.io.IOException;
+import java.net.URI;
+
+public class ExpectingHttpRequestFactory implements AsyncClientHttpRequestFactory, ClientHttpRequestFactory {
```

Comment:
Can't we use the existing implementation?

Reply:
I think it is private. Yeah I know I am already using the private constructor of the server, but this guy here is 10 straight forward lines of code